src/gui/Src/Gui/StructWidget.cpp

```diff
@@ -467,7 +487,42 @@ void StructWidget::changeAddrSlot()
 
 void StructWidget::refreshSlot()
 {
-    typeUpdateWidget();
+    auto doRefresh = [&]
+    {
+        typeUpdateWidget();
+
+        // Update mInsertIndex to the current row index of the selected item
+        if(hasSelection)
+            mInsertIndex = ui->treeWidget->currentIndex().row();
+        else
+            mInsertIndex = -1; // Reset if no selection
+    };
+
+    doRefresh();
+
+    if(!hasSelection || selectedItem->parent() || !DbgIsDebugging())
+        return;
+
+    auto selectedAddr = selectedType.addr + selectedType.offset;
+
+    QTreeWidgetItem* parentItem = selectedItem->parent();
+    mInsertIndex = parentItem ? parentItem->indexOfChild(selectedItem) : ui->treeWidget->indexOfTopLevelItem(selectedItem);
+
+    auto type = selectedItem->data(0, Qt::UserRole).value<TypeDescriptor>();
+
+    delete selectedItem;
+
+    DbgCmdExec(QString("DisplayType %1, %2").arg(type.typeName).arg(ToPtrString(selectedAddr)));
+    doRefresh();

```

Comment:
Isn't this a race? The `DisplayType` command might not be executed at this point yet, so `typeUpdateWidget` will do nothing.

Reply:
If I understand the code correctly the code after the `DbgCmdExec` can be completely removed and everything should still work. The `mInsertIndex` makes sure the new node gets inserted in the right position and `typeUpdateWidget` naturally gets called at the end of the `DisplayType` implementation.